packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -381,7 +387,29 @@ export class Summarizer extends EventEmitter implements ISummarizer {
                         summaryAckHandle,
                         refSequenceNumber,
                         summaryLogger,
-                    ));
+                    ).catch(async (error) => {
+                        // If the error is 404, so maybe the fetched version no longer exists on server. We retry to
+                        // refresh with the latest version one time.
+                        if (isFluidError(error)
+                            && error.errorType === DriverErrorType.fileNotFoundOrAccessDeniedError) {
+                            summaryLogger.sendTelemetryEvent({
+                                eventName: "HandleSummaryAckErrorRetry",
+                                referenceSequenceNumber: refSequenceNumber,
+                                proposalHandle: summaryOpHandle,
+                                ackHandle: summaryAckHandle,
+                            }, error);
+                            await this.internalsProvider.refreshLatestSummaryAck(
```

Comment:
We will react to next ack op and will do proper fetch then. Why do we need to make a call here? It feels like it will double number of calls, no?

Reply:
Actually, we don't have the ability to react to all ack ops because currently we fetch a particular version based on the handle from the ack op. It could happen that, that version does not exist in cache and server(as only first and latest is stored on spo) and so we have nowhere to do in that case. 
I think what we should do is when we get a summary ack, we should refresh with latest snapshot instead of the one specified in ack and then ignore all summary op and acks till that refseq number of latest summary.

Reply:
This also begs the question, on getting a summary acknowledgement, if we are to anyway fetch the latest snapshot, then why not load the summarizer from the latest snapshot always?

Reply:
I agree, but my point is - if we are processing an ack that is old (old as in - summary it references does not exist), then there should be a more recent summary, and thus an ack for it. When we see the ack, we will reload info properly, so maybe we should not do anything on failed summary retrieval.
There might be no ack for recent summary at all (with single-commit summary), so that's the reason to fetch latest summary immediately. That makes sense, but I'd try to see if we can ensure we do not fetch same summary twice - once on failure for previous ack op, and once we see a next ack op. I believe locking helps here, but I'm not sure if we do nothing if we see an ack later on for a summary we just fetched - do we ignore it?

Reply:
My general preference - not to complicate code if it's rare event, and simply bail out and restart summarizer.
Summary code is already very complicated, and the less special cases we have the better. This should, of course, be balanced with perf impact to users and COGS on the server - such expensive bail out works Ok if it's rarely needed.

Reply:
I think ignoring the error should work in both cases. As we will have another summaryAck for the latest version with which we will refresh the state. However in case of single commit summary, we might me missing a summary ack, so in that case we are still fine as the code in `submitSummary` function in container runtime, will refresh the latest state after my previous fix, by calling `refreshLatestSummaryAckFromServer` and we will be fine.